shared/modules/macros.jinja

```diff
@@ -0,0 +1,51 @@
+
+{{# Example usage: sshd_check(default="no", option="Banner", value="/etc/issue") #}}
+{{% macro sshd_check(default, option, value) -%}}
+|-
+    To determine how the SSH daemon's <tt>{{{ option }}}</tt> option is set, run the following command:
+    <pre>$ sudo grep -i {{{ option }}} /etc/ssh/sshd_config</pre>
+    {{% if default == "yes" -%}}
+    If no line, a commented line, or a line indicating the value <tt>{{{ value }}}</tt> is returned, then the required value is set.
+    {{%- endif %}}
+    {{% if default == "no" or default == '' -%}}
+    If a line indicating {{{ value }}} is returned, then the required value is set.
+    {{%- endif %}}
+{{%- endmacro %}}
+
+
+{{# Example usage: iptables_describe_allow(proto="tcp", port=80) #}}
+{{% macro iptables_describe_disallow(proto, port) -%}}
+{{{ iptables_desc(
+	traffic_action="to block",
+	how_to_do="Remove the following line, ensuring that it does not appear the INPUT chain:",
+	proto=proto,
+	port=port,
+) }}}
+{{%- endmacro %}}
+
+
+{{% macro iptables_describe_allow(proto, port) -%}}
+{{{ iptables_desc(
+	traffic_action="to allow",
+	how_to_do="Add the following line, ensuring that it appears before the final LOG and DROP lines for the INPUT chain:",
+	proto=proto,
+	port=port,
+) }}}
+{{%- endmacro %}}
+
+
+{{% macro iptables_desc(traffic_action, how_to_do, proto, port) -%}}
+    To configure <tt>iptables</tt> {{{ traffic_action }}} port {{{ port }}} traffic, one must edit
+    <tt>/etc/sysconfig/iptables</tt> and
+    <tt>/etc/sysconfig/ip6tables</tt> (if IPv6 is in use).
+    {{{ how_to_do }}}:
```

Comment:
This produces 2 semicolons, because both how_to_do options contain semicolon.

Reply:
Nice catch, thanks!